universal-login-contracts/contracts/ERC1077.sol

```diff
@@ -114,8 +117,13 @@ contract ERC1077 is KeyHolder, IERC1077 {
         uint256 startingGas = gasleft();
         bytes memory _data;
         bool success;
-        /* solium-disable-next-line security/no-call-value */
-        (success, _data) = to.call.value(value)(data);
+        if (gasToken == address(0)) {
+            /* solium-disable-next-line security/no-call-value */
+            (success, _data) = to.call.gas(gasleft().sub(ETHER_REFUND_CHARGE)).value(value)(data);
+        } else {
+            /* solium-disable-next-line security/no-call-value */
+            (success, _data) = to.call.gas(gasleft().sub(TOKEN_REFUND_CHARGE)).value(value)(data);
+        }
```

Comment:
Why not simpler?
```solidity
  uint refundGas;
  if (gasToken == address(0)) {
    refundGas = ETHER_REFUND_CHARGE;
  } else {
    refundGas = TOKEN_REFUND_CHARGE;
  }
  /* solium-disable-next-line security/no-call-value */
  (success, _data) = to.call.gas(gasleft().sub(refundGas)).value(value)(data);
```

Now we can extract pure function `refundGas(address gasToken)` and have:
```ts
  /* solium-disable-next-line security/no-call-value */
  (success, _data) = to.call.gas(gasleft().sub(refundGas(gasToken))).value(value)(data);
```